src/libs/suomifi-utils.ts

```diff
@@ -0,0 +1,148 @@
+  service.serviceNames.find(serviceName => serviceName.language === language)?.value
+);
+
+/**
+ * Gets Suomi.fi services summary description for given language
+ * 
+ * @param service service
+ * @param language language
+ * @returns service summary description
+ */
+export const getServiceSummary = (service: Service, language: SupportedLanguages) => (
+  service.serviceDescriptions.find(serviceDdescription =>
+    serviceDdescription.language === language && serviceDdescription.type === "Summary"
+  )?.value
+);
+
+/**
+ * Compares equivalency between Suomi.fi services and TPR departments.  
+ * Returns true if titles are equal and summaries equality is over given treshold (defaults to 80%).
+ * 
+ * @param service service
+ * @param department department
+ * @param language language
+ * @param treshold treshold
+ * @returns Whether services were equivalent enough
+ */
+export const compareServices = (service: Service, department: Department, language: SupportedLanguages, treshold: number = 80) => {
+  const suomifiSummary = getServiceSummary(service, language);
+  const suomifiName = getServiceName(service, language);
+  if (suomifiName === department.title && compareSummaries(suomifiSummary, department.description_short) > treshold) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+/**
+ * Returns whichever summary is shorther and longer than the other
+ * 
+ * @params summary1 summary1
+ * @params summary2 summary2
+ */
+const getLongerAndShortherSummary = (summary1: string[], summary2: string[]) => {
+  if (summary1.length > summary2.length) {
+    return {
+      longerSummary: summary1,
+      shortherSummary: summary2
+    }
+  }
+  
+  return {
+    longerSummary: summary2,
+    shortherSummary: summary1
+  }
+};
+
+/**
+ * Compares summaries from Suomi.fi and TPR word by word and returns percentage of matching words.
+ * 
+ * @param suomifiSummary Summary from Suomi.fi service
+ * @param registrySummary Summary from TPR department
+ * @returns Match in percentages
+ */
+const compareSummaries = (suomifiSummary?: string, registrySummary?: string) => {
+  if (!suomifiSummary || !registrySummary) {
+    return 0;
+  }
+  const suomifiSplit = suomifiSummary.split(" ").map(word => word.replace(/[^a-zA-Z0-9 ]/g, ""));
+  const registrySplit = registrySummary.split(" ").map(word => word.replace(/[^a-zA-Z0-9 ]/g, ""));
+  
+  const { longerSummary, shortherSummary } = getLongerAndShortherSummary(suomifiSplit, registrySplit);
+  
+  let amountOfMatchingWords = 0;
+  for (let i = 0; i < longerSummary.length; i++) {
+    if (longerSummary[i] === shortherSummary[i]) {
+      amountOfMatchingWords++;
+    }
+  }
+  
+  return amountOfMatchingWords / shortherSummary.length * 100;
+};
+
+/**
+ * Does HTTP GET request to given path in Suomi.fi API
+ * 
+ * @param path path
+ * @param pageNumber page number
+ * @returns Response
+ */
+const doRequest = async <T>(path: string, pageNumber?: number) => {
```

Comment:
does this actually perform a list request?

Reply:
Resolved